src/reaktoro_pse/reaktoro_block.py

```diff
@@ -0,0 +1,952 @@
+        ),
+    )
+    CONFIG.declare(
+        "presolve_tolerance",
+        ConfigValue(
+            default=1e-4,
+            domain=float,
+            description="Tolerance for reaktoro pre-solve",
+            doc="""Tolerance for reaktoro pre-solver""",
+        ),
+    )
+    CONFIG.declare(
+        "presolve_epsilon",
+        ConfigValue(
+            default=1e-32,
+            domain=float,
+            description="Tolerance for reaktoro pre-solve",
+            doc="""Defines what is considered to be 0 for ion composition""",
+        ),
+    )
+    CONFIG.declare(
+        "hessian_type",
+        ConfigValue(
+            default="Jt.J",
+            domain=IsInstance((str, HessTypes)),
+            description="Hessian type to use for reaktor gray box",
+            doc="""Hessian type to use, some might provide better stability
+                options (Jt.J, BFGS, BFGS-mod,BFGS-damp, BFGS-ipopt""",
+        ),
+    )
+    CONFIG.declare(
+        "open_species_on_property_block",
+        ConfigValue(
+            default=None,
+            domain=IsInstance((str, list)),
+            description="Registers species to open to optimization, this can help with solvability of some problems",
+            doc="""Registers species (or list of species) to open to optimization and write empty constraint for,
+            this can help with solvability of some problems, but can
+            lead to unexpected results depending on database, activity coefficients, and inputs chosen.
+            This generally should not be left as None""",
+        ),
+    )
+    CONFIG.declare(
+        "open_species_on_speciation_block",
+        ConfigValue(
+            default=None,
+            domain=IsInstance((str, list)),
+            description="Registers species to open to optimization, this can help with solvability of some problems",
+            doc="""Registers species (or list of species) to open to optimization and write empty constraint for,
+            this can help with solvability of some problems, but can
+            lead to unexpected results depending on database, activity coefficients, and inputs chosen.
+            This generally should not be needed""",
+        ),
+    )
+
+    def build(self):
+        super().build()
+        """ configure state"""
+        if self.config.build_speciation_block:
+            """create spectitation block and then property block"""
+            self.speciation_block = Block()
+            self.build_rkt_state(self.speciation_block, speciation_block=True)
+            self.build_rkt_inputs(self.speciation_block, speciation_block=True)
+            self.build_rkt_outputs(self.speciation_block, speciation_block=True)
+            self.build_rkt_jacobian(self.speciation_block)
+            self.build_rkt_solver(self.speciation_block, speciation_block=True)
+            self.build_gray_box(self.speciation_block)
+
+            self.build_rkt_state(
```

Comment:
It looks like most of this is the same as what happens in the `else` branch (the default for `speciation_block` is `False` as far as I can tell).

Thus, I think you can make the `if` branch just for the speciation block, and then just run most of the `else` part for all cases. The only difference I see if whether `speciation_block_built` is True or not, which can be set in the `if` branch.

Reply:
done